cli/filplus.go

```diff
@@ -923,3 +931,438 @@ var filplusSignRemoveDataCapProposal = &cli.Command{
+		smsgs, err := api.MpoolBatchPushMessage(ctx, msgs, nil)
+		if err != nil {
+			return err
+		}
+
+		// wait for msgs to get mined into a block
+		eg := errgroup.Group{}
+		eg.SetLimit(10)
+		for _, msg := range smsgs {
+			m := msg
+			eg.Go(func() error {
+				wait, err := api.StateWaitMsg(ctx, m.Cid(), uint64(cctx.Int("confidence")))
+				if err != nil {
+					return fmt.Errorf("timeout waiting for message to land on chain %s", wait.Message)
+
+				}
+
+				if wait.Receipt.ExitCode.IsError() {
+					return fmt.Errorf("failed to execute message %s: %w", wait.Message, wait.Receipt.ExitCode)
+				}
+				return nil
+			})
+		}
+		return eg.Wait()
+	},
+}
+
+type provInfo struct {
+	addr address.Address
+	id   abi.ActorID
+}
+
+// CreateExtendClaimMsg creates extend message[s] based on the following conditions
+// 1. Extend all claims for a miner ID
+// 2. Extend all claims for multiple miner IDs
+// 3. Extend specified claims for a miner ID
+// 4. Extend specific claims for specific miner ID
+// 5. Extend all claims for a miner ID with different client address (2 messages)
+// 6. Extend all claims for multiple miner IDs with different client address (2 messages)
+// 7. Extend specified claims for a miner ID with different client address (2 messages)
+// 8. Extend specific claims for specific miner ID with different client address (2 messages)
+func CreateExtendClaimMsg(ctx context.Context, api v0api.FullNode, pcm map[verifregtypes9.ClaimId]provInfo, miners []string, wallet address.Address, tmax abi.ChainEpoch, all, assumeYes bool) ([]*types.Message, error) {
+
+	ac, err := api.StateLookupID(ctx, wallet, types.EmptyTSK)
+	if err != nil {
+		return nil, err
+	}
+	w, err := address.IDFromAddress(ac)
+	if err != nil {
+		return nil, fmt.Errorf("converting wallet address to ID: %w", err)
+	}
+
+	wid := abi.ActorID(w)
+
+	head, err := api.ChainHead(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var terms []verifregtypes9.ClaimTerm
+	var newClaims []verifregtypes9.ClaimExtensionRequest
+	rDataCap := big.NewInt(0)
+
+	// If --all is set
+	if all {
+		for _, id := range miners {
+			maddr, err := address.NewFromString(id)
+			if err != nil {
+				return nil, fmt.Errorf("parsing miner %s: %w", id, err)
+			}
+			mid, err := address.IDFromAddress(maddr)
+			if err != nil {
+				return nil, fmt.Errorf("converting miner address to miner ID: %w", err)
+			}
+			claims, err := api.StateGetClaims(ctx, maddr, types.EmptyTSK)
+			if err != nil {
+				return nil, fmt.Errorf("getting claims for miner %s: %w", maddr, err)
+			}
+			for cID, c := range claims {
+				claimID := cID
+				claim := c
+				if claim.TermMax < tmax && claim.TermStart+claim.TermMax > head.Height() {
+					// If client is not same - needs to burn datacap
+					if claim.Client != wid {
+						newClaims = append(newClaims, verifregtypes9.ClaimExtensionRequest{
+							Claim:    claimID,
+							Provider: maddr,
```

Comment:
this will probably have to change to just be the ID

Reply:
Updated